### PR TITLE
feat: support passing additional HTTP headers

### PR DESCRIPTION
**Pull request by z0al**
## Summary

Support passing additional HTTP Headers to (most) commands. For example:

```sh
contentful export -H "x-contentful-enable-alpha-feature: my-feature" ...args
```

## Description

Added `--header/-H` option to the following command namespaces:

- Content Type
- Extension
- Organization
- Space

A header value should be `"key: value"` format. Whitespaces are allowed but will be trimmed. A value that doesn't follow the aforementioned format will be ignored silently. 

## Motivation and Context

Enabling users to pass additional headers can be useful in general but especially when testing out alpha features (e.g. part of EAP program)

## Todos

- [x] Support `--header/-H` option in necessary commands
- [ ] Update docs. Is this supposed to be done manually?
- [x] Fix & add more tests if necessary

P.S. Sorry about the changes set. Most of it caused by running `npm run lint -- --fix`. I had to adjust Prettier rules that are mostly ignored throughout the code to get a smaller changeset (was 133 files before 🤯 )

### CODE REVIEW

**Comment by z0al** [comment]
The `/docs` folder seems out of sync to me, what is the status there? Also, do  I have to update the it manually?

File: lib/cmds/space_cmds/generate_cmds/migration.js [hunk 1]
```diff
@@ -323,21 +329,27 @@ const generateMigrationScript = async function(environment, contentTypes) {
...
 }
 
-const getEnvironment = async function(managementToken, spaceId, environmentId) {
-  const client = await createManagementClient(managementToken)
+const getEnvironment = async function (
+  managementToken,
+  spaceId,
+  environmentId,
+  headers
+) {
```
Comment (damienxy) [inline lib/cmds/space_cmds/generate_cmds/migration.js:351]: I would turn these params into an object now that the argument list is getting long

**Comment by marcolink** [comment]
@z0al great work!
I was just wondering if the arg should be plural e.g. `headers`? The way I understand it, you can also pass a list of headers, right?

**Comment by z0al** [comment]
> @z0al great work!
> I was just wondering if the arg should be plural e.g. `headers`? The way I understand it, you can also pass a list of headers, right?

@marcolink you can pass only one value per option though. To pass multiple values you need to repeat the option (`-H/--header`) e.g.

```sh
cli -H "header1: value" -H "header2: value" ...
```

I followed `curl` example here and don't really have a strong opinion on the matter.

**Comment by marcolink** [comment]
> > @z0al great work!
> > I was just wondering if the arg should be plural e.g. `headers`? The way I understand it, you can also pass a list of headers, right?
> 
> @marcolink you can pass only one value per option though. To pass multiple values you need to repeat the option (`-H/--header`) e.g.
> 
> ```shell
> cli -H "header1: value" -H "header2: value" ...
> ```
> 
> I followed `curl` example here and don't really have a strong opinion on the matter.

Ah, I thought one would pass in a comma-separated list of headers 💡.

Then the current naming is totally fine :)

**Comment by z0al** [comment]
> Ah, I thought one would pass in a comma-separated list of headers bulb.

That will require a bit more sophisticated header parsing than a simple string split :)

**Review by marcolink — APPROVED** [review]
:shipit:

**Pull request merged**